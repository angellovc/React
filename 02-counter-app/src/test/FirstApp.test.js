import React from 'react';
// import { render } from "@testing-library/react";
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import FirstApp from '../FirstApp';

describe('<FisrtApp /> Test', () => {
    // test('Message "Hi, Im Angello" should be displayed', () => {
    //     const greeting = 'Hi, Im Angello';
    //     const {getByText} = render(<FirstApp saludo={greeting} />);
    //     expect(getByText(greeting)).toBeInTheDocument();
    // });

    test('<Must be displayed correctly', () => {
        const greeting = 'Hi, Im Angello';
        const wrapper = shallow(<FirstApp saludo={greeting} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Must show the subtitle sent by props', () => {
        const greeting = 'Hi, Im Angello';
        const subtitle = 'This is another subtitle';
        const wrapper = shallow(
            <FirstApp
                saludo={greeting}
                subtitle={subtitle}
            />
        );
        const textFromParagraph = wrapper.find('p').text();
        expect(textFromParagraph).toBe(subtitle);
    });

})