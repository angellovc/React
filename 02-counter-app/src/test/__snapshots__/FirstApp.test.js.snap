// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FisrtApp /> Test <FirstApp /> must be displayed correctly 1`] = `
<Fragment>
  <h1>
    Hi, Im Angello
  </h1>
  <p>
    im a subtitle
  </p>
</Fragment>
`;

exports[`<FisrtApp /> Test <Must be displayed correctly 1`] = `
<Fragment>
  <h1>
    Hi, Im Angello
  </h1>
  <p>
    im a subtitle
  </p>
</Fragment>
`;
